File: rust/src/model/enums/Scheinuntergruppe.rs
pub struct Scheinuntergruppe<'a>(&'a str);

impl Scheinuntergruppe<'static> {
    pub const AUFTRAGSLEISTUNGEN: Scheinuntergruppe<'static> = Scheinuntergruppe("Auftragsleistungen");
    pub const KONSILIARUNTERSUCHUNG: Scheinuntergruppe<'static> = Scheinuntergruppe("Konsiliaruntersuchung");
    pub const MIT_WEITERBEHANDLUNG: Scheinuntergruppe<'static> = Scheinuntergruppe("MitWeiterbehandlung");
    pub const MUSTER10: Scheinuntergruppe<'static> = Scheinuntergruppe("Muster10");
    pub const MUSTER10A: Scheinuntergruppe<'static> = Scheinuntergruppe("Muster10A");
}

File: java/src/main/java/libldt3/model/enums/Scheinuntergruppe.java
/*
 * Copyright 2016-2023  Christoph Brill &lt;opensource@christophbrill.de&gt;
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package libldt3.model.enums;

/**
 * E009
 */
public enum Scheinuntergruppe {
    Auftragsleistungen("21"),
    Konsiliaruntersuchung("23"),
    /** Mit-/Weiterbehandlung */
    Mit_Weiterbehandlung("24"),
    /** Überweisungsschein für Laboratoriumsuntersuchungen als
     * Auftragsleistung (Muster 10) */
    Muster10("27"),
    /** Anforderungsschein für Laboratoriumsuntersuchungen bei
     * Laborgemeinschaften (Muster 10A) */
    Muster10A("28");

    public final String code;

    Scheinuntergruppe(String code) {
        this.code = code;
    }

}
